internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -673,14 +704,13 @@ func modifyMappedVirtualDisk(
 		return nil
 	case guestrequest.RequestTypeRemove:
 		if mvd.MountPath != "" {
-			err = securityPolicy.EnforceDeviceUnmountPolicy(mvd.MountPath)
-			if err != nil {
-				return errors.Wrapf(err, "unmounting scsi device at %s denied by policy", mvd.MountPath)
+			if mvd.ReadOnly {
+				if err := securityPolicy.EnforceDeviceUnmountPolicy(mvd.MountPath); err != nil {
```

Comment:
Still not clear to me why we are only enforcing policy on readonly SCSI drives.

Reply:
because scratch disks dont go through the device mount/unmount and the only devices that do are readonly.